sdk/requirement.go

```diff
@@ -74,6 +89,15 @@ func (l RequirementList) IsValid() error {
 		return WithStack(ErrInvalidJobRequirementDuplicateHostname)
 	}
 
+	// check that secret requirements are valid regexp
+	for i := range l {
+		if l[i].Type == SecretRequirement {
+			if _, err := regexp.Compile(l[i].Value); err != nil {
+				return NewErrorFrom(ErrInvalidJobRequirement, "requirement of type secret with value %q is not a valid regex: %v", l[i].Value, err)
```

Comment:
did you test how this error is raised to the user on UI and on As Code ?
we have to be sure that it would be clear enough

Reply:
Code updated to run validation for ascode, the error will be displayed like the following example:
"⚠ An error has occurred: Invalid job requirement (from: requirement of type secret with value "..." is not a valid regex: error parsing regexp: invalid escape sequence: `...`).